modal/functions.py

```diff
@@ -471,6 +472,11 @@ def _get_context(self) -> Tuple[_Client, str]:
     @property
     def web_url(self) -> str:
         """URL of a Function running as a web endpoint."""
+        from modal import is_local
+
+        if not is_local():
```

Comment:
we should fix this though (although it's not trivial)

Reply:
Yeah totally. Just a quick thing to at least warn users this is not possible for now.

Reply:
Yep sounds right. I think roughly what has to happen is `AppGetObjects` needs to return protobufs containing additional stuff.